src/SeleniumLibrary/keywords/webdrivertools.py

```diff
@@ -133,86 +136,108 @@ def _geckodriver_log(self):
         return log_file
 
     def create_headless_firefox(self, desired_capabilities, remote_url,
-                                ff_profile_dir, service_log_path=None):
-        options = webdriver.FirefoxOptions()
+                                ff_profile_dir, options=None, service_log_path=None):
+        ff_options = webdriver.FirefoxOptions() if not options else options
         # Can be changed to options.headless = True when minimum Selenium version is 3.12.0 or greater.
-        options.set_headless()
-        return self.create_firefox(desired_capabilities, remote_url, ff_profile_dir, options, service_log_path)
+        ff_options.set_headless()
+        return self.create_firefox(desired_capabilities, remote_url, ff_profile_dir, ff_options, service_log_path)
 
-    def create_ie(self, desired_capabilities, remote_url, service_log_path=None):
+    def create_ie(self, desired_capabilities, remote_url, options=None, service_log_path=None):
         if is_truthy(remote_url):
             defaul_caps = webdriver.DesiredCapabilities.INTERNETEXPLORER.copy()
             desired_capabilities = self._remote_capabilities_resolver(desired_capabilities, defaul_caps)
-            return self._remote(desired_capabilities, remote_url)
-        if self._has_service_log_path(webdriver.Ie):
-            return webdriver.Ie(service_log_path=service_log_path, **desired_capabilities)
-        logger.warn('This version of Selenium does not support service_log_path argument.')
+            return self._remote(desired_capabilities, remote_url, options=options)
+        if self._has_service_log_path(webdriver.Ie) and self._has_options(webdriver.Ie):
+            # service_log_path is supported from Selenium 3.14 onwards
+            # If can be removed when minimum Selenium version is 3.14.0 or greater
+            return webdriver.Ie(options=options, service_log_path=service_log_path, **desired_capabilities)
+        elif not self._has_service_log_path(webdriver.Ie) and self._has_options(webdriver.Ie):
+            # options is supported from Selenium 3.10 onwards
+            # If can be removed when minimum Selenium version is 3.10.0 or greater
+            logger.warn('This version of Selenium does not support service_log_path argument.')
+            return webdriver.Ie(options=options, **desired_capabilities)
+        logger.warn('This version of Selenium does not support options or service_log_path argument.')
```

Comment:
These changes look somewhat unrelated to other changes.

Reply:
Ie and some other browser was difficult to make, because support for Selenium options and service_log_path came in different Selenium version. There, depending on the Selenium version `Ie()` needs to be called with different arguments.

Reply:
Also there was a bug which was fixed in drive by manner, but I do not anymore recall was it this or some other place.